pytorch_lightning/trainer/trainer.py

```diff
@@ -924,6 +925,8 @@ def _maybe_switch_to_iterator_batch_processor(self, model: "pl.LightningModule")
             )
             batch_loop = IteratorBatchProcessor(self, model)
             self.fit_loop.epoch_loop.connect(batch_loop)
+            # FIXME: Move this logic to data_connector after removing `IteratorBatchProcessor`
```

Comment:
Will you do this in this PR or in some follow-ups?

Reply:
follow ups.